src/course-outline/data/slice.js

```diff
@@ -53,18 +55,69 @@ const slice = createSlice({
         ...state.loadingStatus,
         outlineIndexLoadingStatus: payload.status,
       };
+      if (payload.errors) {
+        state.errors = {
+          ...state.errors,
+          outlineIndexApi: payload.errors,
+        };
+      } else {
+        const errors = { ...state.errors };
```

Comment:
Is there a reason you didn't use this way of getting errors from state to update?
```suggestion
        const { errors}  = state;
```

Reply:
You could also use array.filter on the keys to remove the specific error key instead of using delete

Reply:
> Is there a reason you didn't use this way of getting errors from state to update?

@KristinAoki Wanted to create a copy of errors object.

> You could also use array.filter on the keys to remove the specific error key instead of using delete

Since `errors` is an object and not an array, it is easier to use delete no?

Reply:
@xitij2000 I was under the impression that we cannot mutate state items directly due to how redux use to work before. Thanks, the code is not a lot simpler.